algorithms/searching/kmp.py

```diff
@@ -0,0 +1,50 @@
+'''
```

Comment:
This looks great. Can you add the naive implementation in another PR following this same Class hierarchy?

Reply:
sure my friend